### ISSUE TITLE
💅 `noLabelWithoutControl` should allow labeling `button` tags

### ISSUE THREAD

**Issue by MrOxMasTer** [biomejs/biome#4511]
### Environment information

```bash
biome: The term 'biome' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

### Rule name

noLabelWithoutControl

### Playground link

https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABNAGUAbgB1ACAAPQAgACgAKQAgAD0APgAgAHsACgAgACAALwAvACAAYwBvAG4AcwB0ACAAaQBzAEgAeQBkAHIAYQB0AGUAZAAgAD0AIAB1AHMAZQBIAHkAZAByAGEAdABpAG8AbgBDAG8AbgB0AGUAeAB0ACgAKQA7AAoAIAAgAGMAbwBuAHMAdAAgAGkAcwBEAGUAcwBrAHQAbwBwACAAPQAgAHUAcwBlAE0AZQBkAGkAYQBRAHUAZQByAHkAKAAnACgAbQBpAG4ALQB3AGkAZAB0AGgAOgAgADEANAA0ADAAcAB4ACkAJwApADsACgAgACAAYwBvAG4AcwB0ACAAWwBpAHMATwBwAGUAbgBNAGUAbgB1ACwAIABzAGUAdABJAHMATwBwAGUAbgBNAGUAbgB1AF0AIAA9ACAAdQBzAGUAUQB1AGUAcgB5AFMAdABhAHQAZQAoAAoAIAAgACAAIAAnAGkAcwBPAHAAZQBuAE0AZQBuAHUAJwAsAAoAIAAgACAAIABwAGEAcgBzAGUAQQBzAEIAbwBvAGwAZQBhAG4ALgB3AGkAdABoAEQAZQBmAGEAdQBsAHQAKABmAGEAbABzAGUAKQAsAAoAIAAgACkAOwAKAAoAIAAgAGMAbwBuAHMAdAAgAGMAbABvAHMAZQBNAGUAbgB1ACAAPQAgAHUAcwBlAEMAYQBsAGwAYgBhAGMAawAoACgAKQAgAD0APgAgAHMAZQB0AEkAcwBPAHAAZQBuAE0AZQBuAHUAKABmAGEAbABzAGUAKQAsACAAWwBzAGUAdABJAHMATwBwAGUAbgBNAGUAbgB1AF0AKQA7AAoACgAgACAALwAvACAAZABpAHMAYQBiAGwAZQAgAHMAYwByAG8AbABsAGkAbgBnAAoAIAAgAHUAcwBlAEUAZgBmAGUAYwB0ACgAKAApACAAPQA%2BACAAewAKACAAIAAgACAAZABvAGMAdQBtAGUAbgB0AC4AYgBvAGQAeQAuAHMAdAB5AGwAZQAuAG8AdgBlAHIAZgBsAG8AdwBZACAAPQAgAGkAcwBPAHAAZQBuAE0AZQBuAHUAIAA%2FACAAJwBoAGkAZABkAGUAbgAnACAAOgAgACcAYQB1AHQAbwAnADsACgAgACAAfQAsACAAWwBpAHMATwBwAGUAbgBNAGUAbgB1AF0AKQA7AAoACgAgACAALwAvACAAYwBvAG4AcwB0ACAAaQBzAEkAbgBlAHIAdAAgAD0AIABpAHMASAB5AGQAcgBhAHQAZQBkACAAPwAgACgAaQBzAEQAZQBzAGsAdABvAHAAIAA%2FACAAZgBhAGwAcwBlACAAOgAgACEAaQBzAE8AcABlAG4ATQBlAG4AdQApACAAOgAgAGYAYQBsAHMAZQA7AAoAIAAgAGMAbwBuAHMAdAAgAGkAcwBJAG4AZQByAHQAIAA9ACAAaQBzAEQAZQBzAGsAdABvAHAAIAA%2FACAAZgBhAGwAcwBlACAAOgAgACEAaQBzAE8AcABlAG4ATQBlAG4AdQA7AAoACgAgACAAcgBlAHQAdQByAG4AIAAoAAoAIAAgACAAIAA8AD4ACgAgACAAIAAgACAAIAA8AEEAYwB0AGkAbwBuAAoAIAAgACAAIAAgACAAIAAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgBmAGwAZQB4ACAAaQB0AGUAbQBzAC0AYwBlAG4AdABlAHIAIABqAHUAcwB0AGkAZgB5AC0AYwBlAG4AdABlAHIAIAByAG8AdQBuAGQAZQBkAC0AWwA2ADIAcAB4AF0AIABwAC0AMQAuADUAIAB0AGUAeAB0AC0AMgB4AGwAIAB0AGUAeAB0AC0AYgBsAGEAYwBrACAAdAByAGEAbgBzAGkAdABpAG8AbgAtAGMAbwBsAG8AcgBzACAAaABvAHYAZQByADoAYgBnAC0AZwBhAGwAbABlAHIAeQAgAG0AZAA6AGgAaQBkAGQAZQBuACIACgAgACAAIAAgACAAIAAgACAAYQByAGkAYQAtAGwAYQBiAGUAbAA9ACIAbwBwAGUAbgAgAE0AZQBuAHUAIgAKACAAIAAgACAAIAAgACAAIABhAHIAaQBhAC0AaABhAHMAcABvAHAAdQBwAD0AIgBkAGkAYQBsAG8AZwAiAAoAIAAgACAAIAAgACAAIAAgAG8AbgBDAGwAaQBjAGsAPQB7ACgAKQAgAD0APgAgAHMAZQB0AEkAcwBPAHAAZQBuAE0AZQBuAHUAKAB0AHIAdQBlACkAfQAKACAAIAAgACAAIAAgAD4ACgAgACAAIAAgACAAIAAgACAAPABJAGMAbwBuACAAbgBhAG0AZQA9ACIAbQBlAG4AdQAiACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAHMAaQB6AGUALQA2ACIAIAAvAD4ACgAgACAAIAAgACAAIAA8AC8AQQBjAHQAaQBvAG4APgAKACAAIAAgACAAIAAgADwAaQBuAHAAdQB0AAoAIAAgACAAIAAgACAAIAAgAGMAaABlAGMAawBlAGQAPQB7AGkAcwBPAHAAZQBuAE0AZQBuAHUAfQAKACAAIAAgACAAIAAgACAAIABvAG4AQwBoAGEAbgBnAGUAPQB7ACgAZQApACAAPQA%2BACAAcwBlAHQASQBzAE8AcABlAG4ATQBlAG4AdQAoAGUALgB0AGEAcgBnAGUAdAAuAGMAaABlAGMAawBlAGQAKQB9AAoAIAAgACAAIAAgACAAIAAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgBwAGUAZQByACAAaABpAGQAZABlAG4AIgAKACAAIAAgACAAIAAgACAAIAB0AHkAcABlAD0AIgBjAGgAZQBjAGsAYgBvAHgAIgAKACAAIAAgACAAIAAgACAAIABpAGQAPQAiAGIAdQByAGcAZQByACIACgAgACAAIAAgACAAIAAvAD4ACgAgACAAIAAgACAAIAA8AGQAaQB2ACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAGEAYgBzAG8AbAB1AHQAZQAgAGwAZQBmAHQALQAwACAAdABvAHAALQAwACAAegAtADUAMAAgAGgALQBzAGMAcgBlAGUAbgAgAHcALQBmAHUAbABsACAALQB0AHIAYQBuAHMAbABhAHQAZQAtAHkALQBmAHUAbABsACAAcABlAGUAcgAtAGMAaABlAGMAawBlAGQAOgAtAHQAcgBhAG4AcwBsAGEAdABlAC0AeQAtADAAIABtAGQAOgBzAHQAYQB0AGkAYwAgAG0AZAA6AHMAaQB6AGUALQBhAHUAdABvACAAbQBkADoALQB0AHIAYQBuAHMAbABhAHQAZQAtAHkALQAwACAAcABlAGUAcgAtAGMAaABlAGMAawBlAGQAOgBbACYAXwBkAGkAdgBdADoALQB0AHIAYQBuAHMAbABhAHQAZQAtAHkALQAwACIAPgAKACAAIAAgACAAIAAgACAAIAA8AGQAaQB2ACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAGEAYgBzAG8AbAB1AHQAZQAgAGwAZQBmAHQALQAwACAAdABvAHAALQAwACAAegAtADEAMAAgAHcALQBmAHUAbABsACAALQB0AHIAYQBuAHMAbABhAHQAZQAtAHkALQBmAHUAbABsACAAYgBnAC0AdwBoAGkAdABlACAAdAByAGEAbgBzAGkAdABpAG8AbgAtAHQAcgBhAG4AcwBmAG8AcgBtACAAbQBkADoAcwB0AGEAdABpAGMAIABtAGQAOgB3AC0AYQB1AHQAbwAgAG0AZAA6AC0AdAByAGEAbgBzAGwAYQB0AGUALQB5AC0AMAAiAD4ACgAgACAAIAAgACAAIAAgACAAIAAgADwAZABpAHYAIABjAGwAYQBzAHMATgBhAG0AZQA9ACIAYwBvAG4AdABhAGkAbgBlAHIAIABzAGkAegBlAC0AZgB1AGwAbAAgAG0AZAA6AG4AbwAtAGMAbwBuAHQAYQBpAG4AZQByACIAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwAZABpAHYAIABjAGwAYQBzAHMATgBhAG0AZQA9ACIAcgBlAGwAYQB0AGkAdgBlACIAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAPABBAGMAdABpAG8AbgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAGEAcgBpAGEALQBsAGEAYgBlAGwAPQAiAEMAbABvAHMAZQAgAG0AZQBuAHUAIgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgBhAGIAcwBvAGwAdQB0AGUAIAByAGkAZwBoAHQALQAwACAAdABvAHAALQA1ACAAbQB0ADoAdABvAHAALQA4ACAAbQBkADoAaABpAGQAZABlAG4AIgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAGkAZAA9ACIAYgB1AHIAZwBlAHIAIgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwASQBjAG8AbgAgAG4AYQBtAGUAPQAiAGwAdQBjAGkAZABlAC8AeAAiACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAHMAaQB6AGUALQA5ACIAIAAvAD4ACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwALwBBAGMAdABpAG8AbgA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA8AG4AYQB2AD4ACgAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA8AEwAaQBzAHQATQBlAG4AdQAgAGMAbABvAHMAZQBNAGUAbgB1AD0AewBjAGwAbwBzAGUATQBlAG4AdQB9ACAALwA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAA8AC8AbgBhAHYAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgADwALwBkAGkAdgA%2BAAoAIAAgACAAIAAgACAAIAAgACAAIAA8AC8AZABpAHYAPgAKACAAIAAgACAAIAAgACAAIAA8AC8AZABpAHYAPgAKACAAIAAgACAAIAAgACAAIAA8AGwAYQBiAGUAbAAKACAAIAAgACAAIAAgACAAIAAgACAAaAB0AG0AbABGAG8AcgA9ACIAYgB1AHIAZwBlAHIAIgAKACAAIAAgACAAIAAgACAAIAAgACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAGEAYgBzAG8AbAB1AHQAZQAgAHoALQAwACAAcwBpAHoAZQAtAGYAdQBsAGwAIABiAGcALQBiAGwAYQBjAGsALwA1ADAAIABtAGQAOgBoAGkAZABkAGUAbgAiAAoAIAAgACAAIAAgACAAIAAgAC8APgAKACAAIAAgACAAIAAgADwALwBkAGkAdgA%2BAAoAIAAgACAAIAA8AC8APgAKACAAIAApADsACgB9ADsACgA%3D

### Expected result

Label should not have been highlighted, as this is not against W3C rules:

![image](https://github.com/user-attachments/assets/fc80dc43-85ec-4ab8-a661-3ea2270a33ef)
label works the same way on the button

https://validator.w3.org/nu/#textarea

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
Could you provide a more minimal repro? There's a lot of code in that playground.

**Comment by MrOxMasTer** [comment]
> Could you provide a more minimal repro? There's a lot of code in that playground.

https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABNAGUAbgB1ACAAPQAgACgAKQAgAD0APgAgAHsACgAgACAAcgBlAHQAdQByAG4AIAAoAAoAIAAgACAAIAA8AD4ACgAgACAAIAAgACAAIAAgACAAPABiAHUAdAB0AG8AbgAgAHQAeQBwAGUAPQAiAGIAdQB0AHQAbwBuACIAIABpAGQAPQAiAGIAdQByAGcAZQByACIAPgAKACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgACAAIAAgAGIAdQB0AHQAbwBuAAoAIAAgACAAIAAgACAAIAAgADwALwBiAHUAdAB0AG8AbgA%2BAAoAIAAgACAAIAAgACAAIAAgADwAbABhAGIAZQBsAAoAIAAgACAAIAAgACAAIAAgACAAIABoAHQAbQBsAEYAbwByAD0AIgBiAHUAcgBnAGUAcgAiAAoAIAAgACAAIAAgACAAIAAgACAAIABjAGwAYQBzAHMATgBhAG0AZQA9ACIAYQBiAHMAbwBsAHUAdABlACAAegAtADAAIABzAGkAegBlAC0AZgB1AGwAbAAgAGIAZwAtAGIAbABhAGMAawAvADUAMAAgAG0AZAA6AGgAaQBkAGQAZQBuACIACgAgACAAIAAgACAAIAAgACAALwA%2BAAoAIAAgACAAIAA8AC8APgAKACAAIAApADsACgB9ADsACgA%3D

**Comment by dyc3** [comment]
Your right, labeling a button shouldn't cause the rule to trigger.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label
